### PR TITLE
Unify DAM apps

### PR DESCRIPTION
**Pull request by jelz**
This PR unifies DAM (Bynder, Cloudinary) apps using a shared base package (`shared-dam-app`) so they:

- [x] Have exactly the same mechanics
- [x] Have drag and drop
- [x] Get their configuration screen rendered automatically using extension parameters

This PR doesn't include style changes.

### CODE REVIEW

File: marketplace/bynder-assets/src/index.js [hunk 1]
```diff
@@ -1,24 +1,96 @@
...
+    parameters: { ...config },
+    width: 1400
+  });
+
+  return Array.isArray(result) ? result : [];
+}
+
+function isDisabled() {
+  return false;
+}
```
Comment (davidfateh) [inline marketplace/bynder-assets/src/index.js]: Is this function even needed?

File: marketplace/bynder-assets/src/index.js [hunk 2]
```diff
@@ -1,24 +1,96 @@
...
+
+function validateParameters({ bynderURL }) {
+  const hasValidProtocol = bynderURL.startsWith('https://');
+  const isHTMLSafe = ['"', '<', '>'].every(unsafe => !bynderURL.includes(unsafe));
```
Comment (davidfateh) [inline marketplace/bynder-assets/src/index.js]: Is this something we should introduce into other apps or is this specifically due to some functionality in this app?

File: marketplace/cloudinary-assets/src/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    insertHandler: data => sdk.close(data)
+  });
+
+  instance.show();
+
+  sdk.window.updateHeight(window.outerHeight);
+}
+
+async function openDialog(sdk, currentValue, config) {
+  const maxFiles = config.maxFiles - currentValue.length;
```
Comment (davidfateh) [inline marketplace/cloudinary-assets/src/index.js]: If this value is `<= 0` should we actually disable the dialog?

File: marketplace/shared-dam-app/src/AppConfig/AppConfig.spec.tsx [hunk 4]
```diff
@@ -39,23 +40,33 @@ const makeSdkMock = () => ({
   }
 });
 
+const validate = () => null; // Means no error
```
Comment (davidfateh) [inline marketplace/shared-dam-app/src/AppConfig/AppConfig.spec.tsx]: Is this something that needs to be changed?

File: marketplace/bynder-assets/src/index.js (hunk 1 above)
Comment (jelz) [reply marketplace/bynder-assets/src/index.js]: yes, `setup` function from `shared-dam-app` expects interface expressed as `Integration` (see type definitions). There is nothing specific to the Bynder app (as oppesed to Cloudinary with its asset limit) that could disable editing so we always return `false`

File: marketplace/bynder-assets/src/index.js (hunk 2 above)
Comment (jelz) [reply marketplace/bynder-assets/src/index.js]: This is specific to the implementation of this app - see above how we use `bynderURL` parameter directly in HTML.

File: marketplace/cloudinary-assets/src/index.js (hunk 3 above)
Comment (jelz) [reply marketplace/cloudinary-assets/src/index.js]: This is done in `isDisabled`, if it returns `true` it won't be even possible to end up in this place.

File: marketplace/shared-dam-app/src/AppConfig/AppConfig.spec.tsx (hunk 4 above)
Comment (jelz) [reply marketplace/shared-dam-app/src/AppConfig/AppConfig.spec.tsx]: it's a not problematic mock ;)

File: marketplace/bynder-assets/src/index.js (hunk 1 above)
Comment (davidfateh) [reply marketplace/bynder-assets/src/index.js]: Ok thanks for explaining.

**Pull request merged**